pkg/api/dtos/plugins.go

```diff
@@ -19,25 +19,29 @@ type PluginSetting struct {
 	JsonData      map[string]interface{}      `json:"jsonData"`
 	DefaultNavUrl string                      `json:"defaultNavUrl"`
 
-	LatestVersion string                  `json:"latestVersion"`
-	HasUpdate     bool                    `json:"hasUpdate"`
-	State         plugins.PluginState     `json:"state"`
-	Signature     plugins.PluginSignature `json:"signature"`
+	LatestVersion string                        `json:"latestVersion"`
+	HasUpdate     bool                          `json:"hasUpdate"`
+	State         plugins.PluginState           `json:"state"`
+	Signature     plugins.PluginSignatureStatus `json:"signature"`
```

Comment:
rather than having this as a flat object... what about making it nested?  and then use the same object in both the singe value and the list value?

Reply:
Yeah I thought about this - would be cleaner alright. I wanted to have frontend weigh-in and get their opinion before changing

Reply:
Any thoughts on this?  I am nervous about adding more and more to the root element.  There is a long list of more metadata we want to support, so breaking it into clear sections will be helpful

Reply:
It's a fair point @ryantxu, I don't know enough about client side usage though to offer an opinion.

Reply:
I agree with @ryantxu here, question is what properties would we nest? Is this only about Signarture, SignatureType, and SignatureOrg? Nesting other properties touched by this PR, i.e State, are used in few places and nesting those would require some frontend changes. But I assume Ryan's comment refers to signature specific props.

Reply:
I assume all related signature properties, which in this case would be the pre-existing `Signature` field which represents the signature state, and both new `SignatureType` and `SignatureOrg` fields. Maybe something like:

```json
...
"signature": {
  "status": "valid",
  "type"  : "private",
  "org"   : "wbrowne"
}
```

Reply:
@dprokop @ryantxu Are you guys happy to merge as is now (so as not to force a frontend change) with the aim to update this when frontend starts to tie the pieces together?

Reply:
@wbrowne I think it's fine, I think next week I could do some work around front-end updates and then we can update the model on the backend if that's OK with you.